src/main/native/windows/processes-jni.cc

```diff
@@ -128,28 +128,25 @@ class NativeProcess {
     }
   }
 
+  // Wait for this process to exit (or timeout).
   jint WaitFor(jlong timeout_msec) {
-    HANDLE handles[1] = {process_};
     DWORD win32_timeout = timeout_msec < 0 ? INFINITE : timeout_msec;
     jint result;
-    switch (WaitForMultipleObjects(1, handles, FALSE, win32_timeout)) {
-      case 0:
+    switch (WaitForSingleObject(process_, win32_timeout)) {
+      case WAIT_OBJECT_0:
         result = 0;
         break;
 
       case WAIT_TIMEOUT:
         result = 1;
         break;
 
-      case WAIT_FAILED:
-        result = 2;
-        break;
-
+      // Any other case is an error and should be reported back to Bazel.
       default:
-        DWORD err_code = GetLastError();
```

Comment:
I prefer keeping this line where it is, so it's clear that no other API call could have taken place since WaitForSingleObject.

Reply:
Ahhh - now I understand why it was split like that :) Of course, I'll put this back in all places!